node/repo/importmgr/mgr.go

```diff
@@ -98,14 +116,50 @@ func (m *Mgr) Info(id multistore.StoreID) (*StoreMeta, error) {
 	return &sm, nil
 }
 
-func (m *Mgr) Remove(id multistore.StoreID) error {
-	if err := m.mds.Delete(id); err != nil {
-		return xerrors.Errorf("removing import: %w", err)
-	}
-
+func (m *Mgr) Remove(id uint64) error {
 	if err := m.ds.Delete(datastore.NewKey(fmt.Sprintf("%d", id))); err != nil {
 		return xerrors.Errorf("removing import metadata: %w", err)
 	}
 
 	return nil
 }
+
+func (m *Mgr) CARV2FilePathFor(dagRoot cid.Cid) (string, error) {
+	importIDs, err := m.List()
+	if err != nil {
+		return "", xerrors.Errorf("failed to fetch import IDs: %w", err)
+	}
+
+	for _, importID := range importIDs {
+		info, err := m.Info(importID)
+		if err != nil {
+			continue
+		}
+		if info.Labels[LRootCid] == "" {
+			continue
+		}
+		c, err := cid.Parse(info.Labels[LRootCid])
+		if err != nil {
+			continue
```

Comment:
We should probably log out the error here

Reply:
Done.